app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -132,7 +128,9 @@ public void onRequestPermissionsResult(int requestCode,
     }
 
     @Override
-    public void onSaveInstanceState(Bundle savedInstanceState) {
+    protected void onSaveInstanceState(Bundle savedInstanceState) {
+        super.onSaveInstanceState(savedInstanceState);

```

Comment:
I'm guessing that in the Android platform's implementation of `onSaveInstanceState` certain fragment info is committed to the bundle. By overriding `onSaveInstanceState` but not passing the bundle to the `super` implementation, we interrupted that code path, preventing the platform from storing whatever data it usually stores into that bundle.